src/flb_parser.c

```diff
@@ -992,6 +994,21 @@ int flb_parser_do(struct flb_parser *parser, const char *buf, size_t length,
     return -1;
 }
 
+int system_utc_offset(int *tmdiff)
+{
+#ifdef _WIN32
+    // Adjust the sign to match tm_gmtoff
+    *tmdiff = -_timezone;
+    return 0;
+#else
+    time_t currentTime = time(NULL);
```

Comment:
This only works for timezones that don't have DST; you need to calculate the offset for the timestamp of each record, not just capture the current offset at process start time. Different log entries will have different offsets because of DST.